tests/vmi_multus_test.go

```diff
@@ -877,29 +877,116 @@ var _ = Describe("SRIOV", func() {
+		if mac != nil {
+			vmi.Spec.Domain.Devices.Interfaces[0].MacAddress = *mac
+		}
+		tests.StartVmOnNode(vmi, nodeName)
+		Expect(err).ToNot(HaveOccurred())
+
+		// configure the client VMI
+		tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
+		configVMIInterfaceWithSudo(vmi, ifaceName, ipCIDR, "\\$ ")
+		return vmi
+	}
+
+	Context("a virtual machine with one macvtap interface, with a custom MAC address", func() {
+		var serverVMI *v1.VirtualMachineInstance
+		chosenMAC := "de:ad:00:00:be:af"
+		serverCIDR := "192.0.2.102/24"
+		var nodeList *k8sv1.NodeList
+		var nodeName string
+
+		BeforeEach(func() {
+			nodeList = tests.GetAllSchedulableNodes(virtClient)
+			Expect(nodeList.Items).NotTo(BeEmpty(), "schedulable kubernetes nodes must be present")
+			nodeName = nodeList.Items[0].Name
+
+			deleteAllVMsAndBlockUntilTheyAreGone(virtClient, tests.NamespaceTestDefault)
+			serverVMI = createCirrosVMIWithMacvtapStaticIP(virtClient, nodeName, macvtapNetworkName, "eth0", serverCIDR, &chosenMAC)
+		})
+
+		It("should have the specified MAC address reported back via the API", func() {
+			vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(serverVMI.Name, &metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(vmi.Status.Interfaces).To(HaveLen(1))
+			Expect(vmi.Status.Interfaces[0].MAC).To(Equal(chosenMAC))
+		})
+
+		Context("and another virtual machine connected to the same network", func() {
+			var clientVMI *v1.VirtualMachineInstance
+			BeforeEach(func() {
+				clientVMI = createCirrosVMIWithMacvtapStaticIP(virtClient, nodeName, macvtapNetworkName, "eth0", "192.0.2.101/24", nil)
+			})
+
+			It("can communicate with the virtual machine in the same network", func() {
+				pingVirtualMachine(clientVMI, cidrToIP(serverCIDR), "\\$ ")
+			})
+		})
+	})
+})
+
 func cidrToIP(cidr string) string {
 	ip, _, err := net.ParseCIDR(cidr)
 	Expect(err).ToNot(HaveOccurred(), "Should be able to parse IP and prefix length from CIDR")
 	return ip.String()
 }
 
-func configInterface(vmi *v1.VirtualMachineInstance, interfaceName, interfaceAddress, prompt string) {
-	cmdCheck := fmt.Sprintf("ip addr add %s dev %s\n", interfaceAddress, interfaceName)
+func configVMIInterfaceWithSudo(vmi *v1.VirtualMachineInstance, interfaceName, interfaceAddress, prompt string) {
+	configInterface(vmi, interfaceName, interfaceAddress, prompt, "sudo ")
+}
+
+func configInterface(vmi *v1.VirtualMachineInstance, interfaceName, interfaceAddress, prompt string, userModifierPrefix ...string) {
+	setStaticIpCmd := fmt.Sprintf("%sip addr add %s dev %s\n", strings.Join(userModifierPrefix, " "), interfaceAddress, interfaceName)
 	err := tests.CheckForTextExpecter(vmi, []expect.Batcher{
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: prompt},
-		&expect.BSnd{S: cmdCheck},
+		&expect.BSnd{S: setStaticIpCmd},
 		&expect.BExp{R: prompt},
 		&expect.BSnd{S: "echo $?\n"},
 		&expect.BExp{R: "0"},
```

Comment:
Since the ip address itself may contain 0, this test may be flaky.
Please refer to https://github.com/kubevirt/kubevirt/pull/3729

Reply:
I think that after rebasing your expecter fixes, this should be addressed.